Annoyance mute (Magic Trees)

FlamingoChest opened the pull request:
Title says it all

Broooklyn committed:
Add Annoyance Mute plugin

Broooklyn committed:
Add Sire and Obelisk sounds

Broooklyn committed:
Add Random Events

Broooklyn committed:
Update Config

Broooklyn committed:
Update README

Broooklyn committed:
Update to v1.0.1

Broooklyn committed:
Add Tekton meteor

Update to v1.0.2

Broooklyn committed:
Remove random events from instanceof NPC

Broooklyn committed:
Add dense essence chiseling

Broooklyn committed:
Update README.md

Broooklyn committed:
Update to v1.0.3

Broooklyn committed:
Update to 1.0.4

Add mute for Chop Chop! special attack

Broooklyn committed:
Add Nightmare parry and Plank Make

Update to v1.0.5

Broooklyn committed:
Add Scarab Swarm sounds

Update to v1.0.6

Broooklyn committed:
Add config sections

Add configs for Alchemy and Fishing sounds

Broooklyn committed:
SoundEffectID: add Fishing and Alchemy

Make protected

Broooklyn committed:
Create Sounds class

Move integer sets from Plugin to Sounds

Add Alchemy and Fishing

Broooklyn committed:
Update README to v1.1

Broooklyn committed:
Update to v1.1

Broooklyn committed:
Add Prayer Sounds

Broooklyn committed:
Update README.md

Broooklyn committed:
Update Annoyance Mute to v1.2

Broooklyn committed:
Add Pickpocket, Woodcutting, NPC Contact, and Others' Area Sounds

Broooklyn committed:
Update README.md to reflect v1.3

Broooklyn committed:
Update annoyance-mute to v1.3

Broooklyn committed:
Add Moo and String Jewelery

Broooklyn committed:
Update annoyance-mute to v1.4

Broooklyn committed:
Add Cave Horrors

Broooklyn committed:
Bump to 1.4.1

Broooklyn committed:
Add Fossil Island Wyverns

Broooklyn committed:
Move Pet stomp out of Sounds.PETS to prevent muting of olm falling crystals

Broooklyn committed:
Add Heal Other

Broooklyn committed:
Update to 1.4.3

Broooklyn committed:
Add Level Up fireworks

Broooklyn committed:
Add some indents

Broooklyn committed:
Add Jellies, Nail Beasts, and Demons

Broooklyn committed:
Update README.md

Broooklyn committed:
Update plugin descriptions

Broooklyn committed:
Update annoyance-mute to v1.5

Broooklyn committed:
Consolidate Sounds.java into Plugin

Broooklyn committed:
Add Mining, Teleport, Nechryael, Dust Devils, and Spectre sounds

Broooklyn committed:
Update to v1.6

Broooklyn committed:
Add other players' teleport sounds

Broooklyn committed:
Add Pickpocket stun

Broooklyn committed:
Update to v1.6.2

Broooklyn committed:
Add Ice Spells

Broooklyn committed:
Add Teleother

Broooklyn committed:
Add Fishing Explosive

Broooklyn committed:
Add Item Drop

Broooklyn committed:
Add Firemaking

Broooklyn committed:
Update Annoyance Mute to v1.7

Broooklyn committed:
Add Thrall attack sounds

Commit:
Added muting sounds by ID

Broooklyn committed:
Merge pull request #54 from greatmoves/annoyance-mute

Added muting sounds by ID

Broooklyn committed:
Add Fletching

Broooklyn committed:
Prevent Mute Thralls from muting tbow

chunderbolt committed:
Add charge orb spells

Broooklyn committed:
Merge pull request #60 from chunderbolt/annoyance-mute

Add charge orb spells

Broooklyn committed:
Add Amethyst fletching

Broooklyn committed:
Update to v1.8.1

JZomDev committed:
Add Vetion Jr to pet sounds

JZomDev committed:
Merge pull request #113 from JZomDev/annoyance-mute_vetion

Annoyance Mute: Add Vetion Jr

JZomDev committed:
Update getVar to getVarbitValue

getVar was terminally deprecated, replacing with getVarbitValue.

JZomDev commented:
@FlamingoChest can you please rebase this on the current branch of annoyance-mute so the changes are more clear? 

Also be sure you're PRing the merge against the specific branch of annoyance-mute and not `master`

JZomDev committed:
Add Smashing! to annoyance mute

JZomDev committed:
Merge pull request #114 from JZomDev/annoyance-mute

Add Smashing! to annoyance mute

JZomDev committed:
Move to a system that allows the same ID to be muted based on the Sound Effect type.

Types being:
Area sound effect and Sound Effect.

As of right now, only Mining gem rocks is something that is impacted by this change as Gem rocks share the same sound id as the Smashing!

JZomDev committed:
Merge pull request #115 from JZomDev/annoyance-mute-soundeffecttypes

Change Sound ID muting system to check sound effect type too

JZomDev committed:
Update build.gradle

JZomDev committed:
Fix NPE on setUpMutes

Add animID to the SoundEffect to determine if the woosh is local player or another player

Broooklyn committed:
Merge pull request #117

Fix NPE and use Animation ID

JZomDev committed:
Before the check on the animID check made it so that when doing an animation it allowed teleport (only others) still played, whoops.

JZomDev committed:
Merge pull request #121 from JZomDev/annoyance-mute-fix-teleport-others

Fix Teleport (only others)

JZomDev committed:
Mute Shattered Relic Cannon "spin"

JZomDev committed:
Merge pull request #122 from JZomDev/annoyance-mute-shattered-cannon

Mute Shattered Relic Cannon "spin"

JZomDev committed:
Move from using the SoundEffect equals to compare  to filtering the HashSet on shouldMute.

I couldn't figure out how to get the equals() override to behave the way I have it now.

JZomDev committed:
Merge pull request #123 from JZomDev/annoyance-mute-fix-teleports-v2

Move from using the SoundEffect equals to compare  to filtering the H…

JZomDev committed:
Add tests

JZomDev committed:
Merge pull request #124 from JZomDev/annoyance-mute-tests

Add tests

JZomDev committed:
Add Ambient Muting

Add Magic tree mute to Ambient Sound list

JZomDev committed:
Up the version number

JZomDev committed:
Merge pull request #125 from JZomDev/annoyance-mute-magic-trees

Add Magic Trees

JZomDev committed:
Put the ambient sounds to mute within setUpMutes()

JZomDev committed:
Reapply ambient sounds when plugin is turned off

JZomDev committed:
Merge pull request #126 from JZomDev/annoyance-mute-magic-trees-setupmutes

Move ambient muting to setUpMutes()

JZomDev committed:
Update build.gradle

Remove 

    implementation 'org.projectlombok:lombok:1.18.20'

JZomDev commented:
Added in https://github.com/Broooklyn/runelite-external-plugins/pull/125
